pkg/hvs/controllers/flavortemplate_controller.go

```diff
@@ -0,0 +1,263 @@
+	defaultLog.Trace("controllers/flavortemplate_controller:validateQueryParameter() Entering")
+	defer defaultLog.Trace("controllers/flavortemplate_controller:validateQueryParameter() Leaving")
+
+	var included bool
+	if len(includeDeleted) > 0 && includeDeleted != "" {
+		if strings.EqualFold(includeDeleted, "true") {
+			included = true
+		} else if strings.EqualFold(includeDeleted, "false") {
+			included = false
+		} else {
+			defaultLog.Error("controllers/flavortemplate_controller:validateQueryParameter() Invalid query parameter given")
+			return false, errors.New("Invalid query parameter given")
+		}
+	}
+	return included, nil
+}
+
+func (ftc FlavorTemplateController) Search(w http.ResponseWriter, r *http.Request) (interface{}, int, error) {
+	defaultLog.Trace("controllers/flavortemplate_controller:Search() Entering")
+	defer defaultLog.Trace("controllers/flavortemplate_controller:Search() Leaving")
+
+	includeDeleted := r.URL.Query().Get("include_deleted")
+
+	included, err := validateQueryParameter(includeDeleted)
+	if err != nil {
+		defaultLog.WithError(err).Error("controllers/flavortemplate_controller:Search() Invalid query parameter given")
+		return false, http.StatusBadRequest, &commErr.ResourceError{Message: "Invalid query parameter given"}
+	}
+
+	//call store function to retrieve all available templates from DB.
+	flavorTemplates, err := ftc.Store.Search(included)
+	if err != nil {
+		defaultLog.WithError(err).Error("controllers/flavortemplate_controller:Search() Error retrieving all flavor templates")
+		return nil, http.StatusInternalServerError, &commErr.ResourceError{Message: err.Error()}
+	}
+
+	return flavorTemplates, http.StatusOK, nil
+}
+
+func (ftc FlavorTemplateController) Delete(w http.ResponseWriter, r *http.Request) (interface{}, int, error) {
+	defaultLog.Trace("controllers/flavortemplate_controller:Delete() Entering")
+	defer defaultLog.Trace("controllers/flavortemplate_controller:Delete() Leaving")
+
+	templateId := uuid.MustParse(mux.Vars(r)["id"])
+	defaultLog.Debug("controllers/flavortemplate_controller:Delete() ID : ", templateId)
+
+	//call store function to delete template from DB.
+	if err := ftc.Store.Delete(templateId); err != nil {
+		defaultLog.WithError(err).Error("controllers/flavortemplate_controller:Delete() Error delete flavor templates")
+		return nil, http.StatusInternalServerError, &commErr.ResourceError{Message: err.Error()}
+	}
+
+	return nil, http.StatusNoContent, nil
+
+}
+
+func getFlavorTemplateCreateReq(r *http.Request) (hvs.FlavorTemplate, error) {
+
+	defaultLog.Trace("controllers/flavortemplate_controller:getFlavorTemplateCreateReq() Entering")
+	defer defaultLog.Trace("controllers/flavortemplate_controller:getFlavorTemplateCreateReq() Leaving")
+
+	var CreateFlavorTemplateReq hvs.FlavorTemplate
+	if r.Header.Get("Content-Type") != constants.HTTPMediaTypeJson {
+		secLog.Error("controllers/flavortemplate_controller:getFlavorTemplateCreateReq() Invalid Content-Type")
```

Comment:
This returns only http error which need to be returned to response, since no file and package name should be in response this cannot be logged in 'create' method so this error logged here.